src/main/java/io/jenkins/plugins/checks/github/GitHubChecksContext.java

```diff
@@ -108,4 +111,37 @@ protected boolean hasValidCredentials(final FilteredLog logger) {
     private Optional<GitHubAppCredentials> findGitHubAppCredentials(final String credentialsId) {
         return getScmFacade().findGitHubAppCredentials(getJob(), credentialsId);
     }
+
+    Optional<Long> getId(final String name) {
+        return getAction(name).map(GitHubChecksAction::getId);
+    }
+
+    // New API method
+    ChecksConclusion getConclusion(final String name) {
+        return getAction(name).map(GitHubChecksAction::getConclusion).orElse(ChecksConclusion.NONE);
+    }
+
+    // Implementation using invisible actions
+    private Optional<GitHubChecksAction> getAction(final String name) {
+        return job.getActions(GitHubChecksAction.class)
+                .stream()
+                .filter(a -> a.getName().equals(name))
+                .findFirst();
+    }
+
+    // Implementation fetching info on request
+    private Optional<GitHubChecksAction> getAction(final GitHub gitHub, final String name) throws IOException {
+        return gitHub.getRepository(getRepository())
+                .getCheckRuns(getHeadSha())
+                .toList()
```

Comment:
I'm sure there's a better way of doing this. The github API certainly supports searching by name, but I'm not sure that's in the hub4j/github-api api